packages/hash/subgraph/src/types/edge/outward-edge-alias.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * A collection of 'aliases' which describe various variants of outward edges in more accessible-forms
+ */
+import { KnowledgeGraphOutwardEdge } from "../edge";
+import { EntityIdAndTimestamp } from "../identifier";
+
+/** @todo - is there a way to have TS force us to make this always satisfy `KnowledgeGraphOutwardEdge`? */
+export type OutwardLinkEdge = {
+  reversed: true;
+  kind: "HAS_LEFT_ENTITY";
+  rightEndpoint: EntityIdAndTimestamp;
+};
+
+export const isOutwardLinkEdge = (
+  outwardEdge: KnowledgeGraphOutwardEdge,
+): outwardEdge is OutwardLinkEdge => {
+  return outwardEdge.kind === "HAS_LEFT_ENTITY" && outwardEdge.reversed;
+};
+
+/** @todo - is there a way to have TS force us to make this always satisfy `KnowledgeGraphOutwardEdge`? */
+export type HasLeftEntityEdge = {
+  reversed: false;
+  kind: "HAS_RIGHT_ENDPOINT";
+  rightEndpoint: EntityIdAndTimestamp;
+};
+
+export const isHasLeftEntityEdge = (
+  outwardEdge: KnowledgeGraphOutwardEdge,
+): outwardEdge is HasLeftEntityEdge => {
+  return outwardEdge.kind === "HAS_RIGHT_ENTITY" && !outwardEdge.reversed;
+};
+
+/** @todo - is there a way to have TS force us to make this always satisfy `KnowledgeGraphOutwardEdge`? */
+export type IncomingLinkEdge = {
+  reversed: true;
+  kind: "HAS_RIGHT_ENDPOINT";
+  rightEndpoint: EntityIdAndTimestamp;
+};
+
+export const isIncomingLinkEdge = (
+  outwardEdge: KnowledgeGraphOutwardEdge,
+): outwardEdge is IncomingLinkEdge => {
+  return outwardEdge.kind === "HAS_RIGHT_ENTITY" && outwardEdge.reversed;
+};
```

Comment:
**Blocking:**
```suggestion
/** @todo - is there a way to have TS force us to make this always satisfy `KnowledgeGraphOutwardEdge`? */
export type IncomingLinkEdge = {
  reversed: true;
  kind: "HAS_RIGHT_ENTITY";
  rightEndpoint: EntityIdAndTimestamp;
};

export const isIncomingLinkEdge = (
  outwardEdge: KnowledgeGraphOutwardEdge,
): outwardEdge is IncomingLinkEdge => {
  return outwardEdge.kind === "HAS_RIGHT_ENTITY" && outwardEdge.reversed;
};
```

Reply:
Damn good spot, I even checked linting and it passed.. I wonder what happened.

Noticed some other problems, fixed it in [Fix edges](https://github.com/hashintel/hash/pull/1438/commits/10c8b6c75a0c3ece4214fb97e5e645f2618cf540)